src/test/regress/sql/multi_complex_expressions.sql

```diff
@@ -158,3 +158,64 @@ SELECT count(*) FROM lineitem JOIN orders ON l_orderkey = o_orderkey
 -- Check that we make sure local joins are between columns only.
 
 SELECT count(*) FROM lineitem, orders WHERE l_orderkey + 1 = o_orderkey;
+
+-- Check that we can issue limit/offset queries
+
+-- Simple LIMIT/OFFSET with ORDER BY
+SELECT o_orderkey FROM orders ORDER BY o_orderkey LIMIT 10 OFFSET 20;
+
+-- LIMIT/OFFSET with a subquery
+SET citus.task_executor_type TO 'task-tracker';
+SELECT 
+	customer_keys.o_custkey,
+	SUM(order_count) AS total_order_count 
+FROM 
+	(SELECT o_custkey, o_orderstatus, COUNT(*) AS order_count 
+	 FROM orders GROUP BY o_custkey, o_orderstatus ) customer_keys
+GROUP BY 
+	customer_keys.o_custkey
+ORDER BY 
+	customer_keys.o_custkey DESC
+LIMIT 10 OFFSET 20;
+
+SET citus.task_executor_type TO 'real-time';
+
+-- Ensure that we push down LIMIT and OFFSET properly
+SET client_min_messages TO 'debug1';
+
+CREATE TEMP TABLE temp_limit_test_1 AS
+SELECT o_custkey FROM orders LIMIT 10 OFFSET 15;
+
+CREATE TEMP TABLE temp_limit_test_2 AS
+SELECT o_custkey FROM orders GROUP BY o_custkey LIMIT 10 OFFSET 15;
+
+CREATE TEMP TABLE temp_limit_test_3 AS
+SELECT o_custkey FROM orders GROUP BY o_custkey ORDER BY o_custkey LIMIT 10 OFFSET 15;
+
+-- Order by aggregate -> No push-down
+CREATE TEMP TABLE temp_limit_test_4 AS
+SELECT o_custkey, COUNT(*) AS ccnt FROM orders GROUP BY o_custkey ORDER BY ccnt DESC LIMIT 10 OFFSET 15;
+
+-- Has aggregate but no order-by -> No push-down
+CREATE TEMP TABLE temp_limit_test_5 AS
+SELECT o_custkey, COUNT(*) AS ccnt FROM orders GROUP BY o_custkey LIMIT 10 OFFSET 15;
+
+RESET client_min_messages;
+DROP TABLE temp_limit_test_1;
```

Comment:
I generally tend to favor cleaning the mess asap but if you say so, let them go away 👍